ietf/community/forms.py

```diff
@@ -38,8 +38,8 @@ def restrict_state(state_type, slug=None):
                 f.initial = f.queryset[0].pk
                 f.widget = forms.HiddenInput()
 
-        if rule_type in ['group', 'group_rfc', 'area', 'area_rfc']:
-            restrict_state("draft", "rfc" if rule_type.endswith("rfc") else "active")
+        if rule_type in ['group', 'group_rfc', 'area', 'area_rfc', 'group_exp']:
+            restrict_state("draft", "rfc" if rule_type.endswith("rfc") else "expired" if rule_type.endswith("exp") else "active")
```

Comment:
This will be hard for future contributors to trust (what's the precedence of each if?). Requires a moderate python-steeping to know.
I think it would be better to make this more verbose and split the computation of the wanted state slug to multiple lines.

Reply:
OK, will try to split.

Reply:
I made the changes you requested - splited the complex condition into two "if" statements and added the tests for both community and group tests (the latter is needed because of some post-filtering, which doesn't show expired drafts if they are dead). I also fixed a logical error - extraneous check whether the "draft" slug is in "auth-rm" or "ietf-rm" (as far as I understand this can never happen when the "draft" is "expired"). Correct me if I'm wrong - the states of the document are cumbersome and I didn't find clear explanation of their relations. I also fixed the last minute error when I renamed auto-generated migration files and forgot to also change "dependencies' in the latter of them.  I think the PR is ready for merging.